mne/filter.py

```diff
@@ -2236,8 +2236,15 @@ def _filt_check_picks(info, picks, h_freq, l_freq):
     # This will pick *all* data channels
     picks = _picks_to_idx(info, picks, 'data_or_ica', exclude=())
     if h_freq is not None or l_freq is not None:
-        data_picks = _picks_to_idx(info, None, 'data_or_ica', exclude=())
-        if np.in1d(data_picks, picks).all():
+        try:
+            data_picks = _picks_to_idx(info, None, 'data_or_ica', exclude=())
+        except ValueError:  # what we throw when there are no data chs
+            data_picks = np.array([], int)
```

Comment:
it would be too painful @larsoner to have _picks_to_idx return `np.array([], int)` 

note that `_picks_to_idx ` has an `allow_empty` parameter that suggests it should be ok if set to True.

Reply:
It might be painful because later in the string function some stuff happens with checking if lengths are zero, but I'll try it to see if it's better